src/converse-chatview.js

```diff
@@ -354,6 +354,9 @@
                             'extra_classes': this.getExtraMessageClasses(attrs)
                         })
                     ));
+                    //Where the argument "attrs" come from? Could "attrs" have information from onMessage?
+                    //Could be possible (and would it be ok) to place a button instead of .text(text)?
+                    //Making the button to show the text set here onclick?
```

Comment:
Use `grep` to look at all the places from where `renderMessage` called. You'll see it's called by `showMessage` and `insertMessage` and they both also accept `attrs`.

So then you can check where and how they are called.

`showMessage` is for example called here: https://github.com/jcbrand/converse.js/blob/master/src/converse-chatview.js#L414

But also in other places.

Eventually you'll see that `attrs` are the attributes on the `Message` Backbone.Model.
And that message model is created here:
https://github.com/jcbrand/converse.js/blob/master/src/converse-core.js#L1420

And from that we can see that attributes of a message are determined by `getMessageAttributes`
Here: https://github.com/jcbrand/converse.js/blob/master/src/converse-core.js#L1369